src/components/modals/TransactionAdvanceDetails.js

```diff
@@ -57,44 +57,40 @@ class TransactionAdvanceDetailsComponent extends PureComponent<Props, State> {
 
   openUrl = () => {
     const { url } = this.props
-    if (url) {
-      if (Platform.OS === 'ios') {
-        return SafariView.isAvailable()
-          .then(SafariView.show({ url }))
-          .catch(error => {
-            Linking.openURL(url)
-            console.log(error)
-          })
-      }
-      Linking.canOpenURL(url).then(supported => {
-        if (supported) {
+    if (url == null) return
+    if (Platform.OS === 'ios') {
+      return SafariView.isAvailable()
+        .then(SafariView.show({ url }))
+        .catch(error => {
           Linking.openURL(url)
-        }
-      })
+          console.log(error)
+        })
     }
+    Linking.canOpenURL(url).then(supported => {
+      if (supported) {
+        Linking.openURL(url)
+      }
+    })
   }
 
   openProveUrl = () => {
     const { txid, txSecret } = this.props.transaction
     const recipientAddress = this.getRecipientAddress()
-    // Early return to satisfy flow. Button isn't visible without all params present.
-    if (!recipientAddress || !txid || !txSecret) return
+    if (recipientAddress === '' || txSecret == null) return
```

Comment:
What about checking for the 'txid'?
And also you need to check that 'txSecret' and 'txid' are not empty strings.